web/src/composebox_typeahead.ts

```diff
@@ -591,11 +591,20 @@ export function get_pm_people(query: string): (UserGroupPillData | UserPillData)
         filter_groups_for_guests: true,
     };
     const suggestions = get_person_suggestions(query, opts);
+    const current_user_ids = compose_pm_pill.get_user_ids();
+    const my_user_id = people.my_current_user_id();
     // We know these aren't mentions because `want_broadcast` was `false`.
     // TODO: In the future we should separate user and mention so we don't have
     // to do this.
     const user_suggestions: (UserGroupPillData | UserPillData)[] = [];
     for (const suggestion of suggestions) {
+        if (
```

Comment:
I think this would benefit from a comment for why we do this in this particular situation.